### PR TITLE
added support to create backup out of automated backups

### PR DESCRIPTION
**Pull request by ghost**
What
---
added support to create backup out of automated backups

*Description of changes:*

- Added support to create snapshot out of automated backups, controlled by flag `UseAutomatedBackup`
- Added `LambdaCWLogRetention` parameter support to control log group expiration
- Tuned copy-snapshot threshold to raise alert

### CODE REVIEW

File: lambda/snapshots_tool_utils.py [hunk 1]
```diff
@@ -373,3 +375,97 @@ def search_tag_copied(response):
 
     return False
 
+
+def get_all_automated_snapshots(client):
+    global _AUTOMATED_BACKUP_LIST
```
Comment (gumdelli) [inline lambda/snapshots_tool_utils.py:380]: Do we need a global variable for this?

File: lambda/snapshots_tool_utils.py [hunk 2]
```diff
@@ -373,3 +375,97 @@ def search_tag_copied(response):
...
+            'describe_db_cluster_snapshots',
+            'DBClusterSnapshots',
+            SnapshotType='automated',
+        )
+        _AUTOMATED_BACKUP_LIST = response['DBClusterSnapshots']
+
+    return _AUTOMATED_BACKUP_LIST
+
+
+def avail_db_snapshot(
```
Comment (gumdelli) [inline lambda/snapshots_tool_utils.py]: Another style comment: this function signature uses new lines for arguments where as the existing functions have a different format, lets maintain the same style as the original code

File: lambda/snapshots_tool_utils.py (hunk 1 above)
Comment (ghost) [reply lambda/snapshots_tool_utils.py:380]: yep, to cache api calls

**Review by ericgay — APPROVED** [review]
all good, looked at 2 last commits post approval

**Pull request merged**